src/webui.c

```diff
@@ -175,6 +196,19 @@ process_request(struct evhttp_request *req, void *arg)
         return;
     }
 
+    char *wn = strstr(url, "/worker/");
+    if ( wn != NULL)
+    {
+        char rig_id[MAX_RIG_ID] = {0};
+        wn += 8;
+        size_t wn_sz = 0;
+        wn = evhttp_uridecode(wn, 1, &wn_sz);
+        if (wn_sz <= MAX_RIG_ID-1 && wn_sz != 0)
+            strncpy(rig_id, wn, MAX_RIG_ID-1);
+        send_json_worker(req, arg, rig_id);
+        return;
+    }
```

Comment:
`char *wn = NULL;` at the top with the others then:

```
if ((wn = strstr(url, "/worker/")))
{
    wn += 8;
    wn = evhttp_uridecode(wn, 1, NULL);
    send_json_worker(req, arg, rig_id);
    return;
}
```

However, really all the parsing should occur in `send_json_worker(req, arg)`, as that's the purpose of these `send_json_*(req, arg)` methods, i.e. take a request, parse, and dump a response.